### ISSUE TITLE
Exception occurred in `FastAIV2PruningCallback` when calling event `after_fit`  

### ISSUE THREAD

**Issue by zmce2018** [optuna/optuna#3615]
### Environment

os             : Windows-10-10.0.19043-SP0
python         : 3.8.13
tsai           : 0.3.1
fastai         : 2.6.3
fastcore       : 1.4.3
optuna         : 2.10.0
torch          : 1.11.0
device         : 1 gpu (['NVIDIA GeForce GTX 1080 Ti'])
cpu cores      : 12
RAM            : 31.95 GB
GPU memory     : [11.0] GB

### Description

I have encountered the Exception occurred in `FastAIV2PruningCallback` when calling event `after_fit` when I was doing fine-tuning. I don't know what is the cause of this exception. 

### Error messages, stack traces, or logs

```shell
[I 2022-06-02 12:27:14,897] Trial 59 finished with value: 0.0494580939412117 and parameters: {'num_filters': 128, 'kernel_size': 4, 'dropout_rate': 0.1, 'learning_rate': 0.004897834628075343}. Best is trial 59 with value: 0.0494580939412117.
[I 2022-06-02 12:27:19,409] Trial 60 pruned. Exception occured in `FastAIV2PruningCallback` when calling event `after_fit`:
	Trial was pruned at epoch 1.
[I 2022-06-02 12:27:21,850] Trial 61 pruned. Exception occured in `FastAIV2PruningCallback` when calling event `after_fit`:
	Trial was pruned at epoch 0.
[I 2022-06-02 12:27:26,604] Trial 62 pruned. Exception occured in `FastAIV2PruningCallback` when calling event `after_fit`:
	Trial was pruned at epoch 1.
[I 2022-06-02 12:27:29,075] Trial 63 pruned. Exception occured in `FastAIV2PruningCallback` when calling event `after_fit`:
	Trial was pruned at epoch 0.
[I 2022-06-02 12:27:31,454] Trial 64 pruned. Exception occured in `FastAIV2PruningCallback` when calling event `after_fit`:
	Trial was pruned at epoch 0.
[I 2022-06-02 12:27:44,448] Trial 65 pruned. Exception occured in `FastAIV2PruningCallback` when calling event `after_fit`:
	Trial was pruned at epoch 5.
[I 2022-06-02 12:27:58,630] Trial 66 pruned. Exception occured in `FastAIV2PruningCallback` when calling event `after_fit`:
	Trial was pruned at epoch 5.
[I 2022-06-02 12:28:03,124] Trial 67 pruned. Exception occured in `FastAIV2PruningCallback` when calling event `after_fit`:
	Trial was pruned at epoch 1.
[I 2022-06-02 12:28:07,816] Trial 68 pruned. Exception occured in `FastAIV2PruningCallback` when calling event `after_fit`:
	Trial was pruned at epoch 1.
[I 2022-06-02 12:28:10,270] Trial 69 pruned. Exception occured in `FastAIV2PruningCallback` when calling event `after_fit`:
	Trial was pruned at epoch 0.
[I 2022-06-02 12:28:25,123] Trial 70 pruned. Exception occured in `FastAIV2PruningCallback` when calling event `after_fit`:
	Trial was pruned at epoch 6.
[I 2022-06-02 12:28:27,376] Trial 71 pruned. Exception occured in `FastAIV2PruningCallback` when calling event `after_fit`:
	Trial was pruned at epoch 0.
[I 2022-06-02 12:28:29,649] Trial 72 pruned. Exception occured in `FastAIV2PruningCallback` when calling event `after_fit`:
	Trial was pruned at epoch 0.
[I 2022-06-02 12:28:31,924] Trial 73 pruned. Exception occured in `FastAIV2PruningCallback` when calling event `after_fit`:
	Trial was pruned at epoch 0.
[I 2022-06-02 12:28:34,108] Trial 74 pruned. Exception occured in `FastAIV2PruningCallback` when calling event `after_fit`:
	Trial was pruned at epoch 0.
[I 2022-06-02 12:28:40,597] Trial 75 pruned. Exception occured in `FastAIV2PruningCallback` when calling event `after_fit`:
	Trial was pruned at epoch 2.
[I 2022-06-02 12:28:42,995] Trial 76 pruned. Exception occured in `FastAIV2PruningCallback` when calling event `after_fit`:
	Trial was pruned at epoch 0.
[I 2022-06-02 12:28:45,234] Trial 77 pruned. Exception occured in `FastAIV2PruningCallback` when calling event `after_fit`:
	Trial was pruned at epoch 0.
[I 2022-06-02 12:29:00,103] Trial 78 pruned. Exception occured in `FastAIV2PruningCallback` when calling event `after_fit`:
	Trial was pruned at epoch 6.
[I 2022-06-02 12:29:02,383] Trial 79 pruned. Exception occured in `FastAIV2PruningCallback` when calling event `after_fit`:
	Trial was pruned at epoch 0.
[I 2022-06-02 12:29:06,770] Trial 80 pruned. Exception occured in `FastAIV2PruningCallback` when calling event `after_fit`:
	Trial was pruned at epoch 1.
[I 2022-06-02 12:29:23,999] Trial 81 pruned. Exception occured in `FastAIV2PruningCallback` when calling event `after_fit`:
	Trial was pruned at epoch 7.
[I 2022-06-02 12:29:26,227] Trial 82 pruned. Exception occured in `FastAIV2PruningCallback` when calling event `after_fit`:
	Trial was pruned at epoch 0.
[I 2022-06-02 12:29:47,780] Trial 83 finished with value: 0.05173031985759735 and parameters: {'num_filters': 64, 'kernel_size': 3, 'dropout_rate': 0.2, 'learning_rate': 0.003884983537102038}. Best is trial 59 with value: 0.0494580939412117.
[I 2022-06-02 12:29:50,022] Trial 84 pruned. Exception occured in `FastAIV2PruningCallback` when calling event `after_fit`:
	Trial was pruned at epoch 0.
[I 2022-06-02 12:29:57,044] Trial 85 pruned. Exception occured in `FastAIV2PruningCallback` when calling event `after_fit`:
	Trial was pruned at epoch 2.
[I 2022-06-02 12:29:59,252] Trial 86 pruned. Exception occured in `FastAIV2PruningCallback` when calling event `after_fit`:
	Trial was pruned at epoch 0.
[I 2022-06-02 12:30:03,607] Trial 87 pruned. Exception occured in `FastAIV2PruningCallback` when calling event `after_fit`:
	Trial was pruned at epoch 1.
[I 2022-06-02 12:30:05,982] Trial 88 pruned. Exception occured in `FastAIV2PruningCallback` when calling event `after_fit`:
	Trial was pruned at epoch 0.
[I 2022-06-02 12:30:12,387] Trial 89 pruned. Exception occured in `FastAIV2PruningCallback` when calling event `after_fit`:
	Trial was pruned at epoch 2.
[I 2022-06-02 12:30:14,672] Trial 90 pruned. Exception occured in `FastAIV2PruningCallback` when calling event `after_fit`:
	Trial was pruned at epoch 0.
[I 2022-06-02 12:30:16,839] Trial 91 pruned. Exception occured in `FastAIV2PruningCallback` when calling event `after_fit`:
	Trial was pruned at epoch 0.
[I 2022-06-02 12:30:19,072] Trial 92 pruned. Exception occured in `FastAIV2PruningCallback` when calling event `after_fit`:
	Trial was pruned at epoch 0.
[I 2022-06-02 12:30:21,341] Trial 93 pruned. Exception occured in `FastAIV2PruningCallback` when calling event `after_fit`:
	Trial was pruned at epoch 0.
[I 2022-06-02 12:30:25,703] Trial 94 pruned. Exception occured in `FastAIV2PruningCallback` when calling event `after_fit`:
	Trial was pruned at epoch 1.
[I 2022-06-02 12:30:28,170] Trial 95 pruned. Exception occured in `FastAIV2PruningCallback` when calling event `after_fit`:
	Trial was pruned at epoch 0.
[I 2022-06-02 12:30:32,650] Trial 96 pruned. Exception occured in `FastAIV2PruningCallback` when calling event `after_fit`:
	Trial was pruned at epoch 1.
[I 2022-06-02 12:30:34,843] Trial 97 pruned. Exception occured in `FastAIV2PruningCallback` when calling event `after_fit`:
	Trial was pruned at epoch 0.
[I 2022-06-02 12:30:39,261] Trial 98 pruned. Exception occured in `FastAIV2PruningCallback` when calling event `after_fit`:
	Trial was pruned at epoch 1.
[I 2022-06-02 12:30:41,721] Trial 99 pruned. Exception occured in `FastAIV2PruningCallback` when calling event `after_fit`:
	Trial was pruned at epoch 0.
```
```


### Steps to reproduce

```python
# python code

from optuna.integration import FastAIPruningCallback

def objective(trial:optuna.Trial):
    
    # Define search space here. More info here https://optuna.readthedocs.io/en/stable/tutorial/10_key_features/002_configurations.html
    nf = trial.suggest_categorical('num_filters', [16,32,64,128]) # search through all categorical values in the provided list
    ks = trial.suggest_categorical('kernel_size', [3,4,5,6]) 
    dropout_rate = trial.suggest_float("dropout_rate", 0.0, 0.5, step=.1)
    learning_rate = trial.suggest_float("learning_rate", 1e-5, 1e-2, log=True)  # search through all float values between 0.0 and 0.5 in log increment steps
    


    learn = TSRegressor(X, y, splits=splits, bs=[64, 128], batch_tfms=batch_tfms,
                         arch=FCNPlus, loss_func = mae,arch_config={'layers':[nf,nf,nf],"kss":[ks,ks,ks],"fc_dropout":dropout_rate},
                         metrics=mae, cbs=FastAIPruningCallback(trial),device='cuda')


    with ContextManagers(mgrs=[learn.no_logging(), learn.no_bar()]): # [Optional] this prevents fastai from printing anything during training
        learn.fit_one_cycle(10, lr_max=learning_rate)

    # Return the objective value
    return learn.recorder.values[-1][1] # return the validation loss value of the last epoch 

study = optuna.create_study(direction='minimize')
study.optimize(objective, n_trials=100)
```

**Comment by nzw0301** [comment]
Hi, could you share minimal reproducible codes with us?

**Comment by zmce2018** [comment]
Hi @nzw0301 

Thank you for your reply. Please find the following code. For the code to be run properly, you might need to install tsai. 

```
from tsai.all import *
import optuna
import numpy as np
#my_setup(optuna)


set_seed(77, True)

'''
y_train = np.load('y_train.npy', mmap_mode='r+')
x_train = np.load('x_train.npy', mmap_mode='r+')
x_valid= np.load('x_valid.npy', mmap_mode='r+')
y_valid= np.load('y_valid.npy', mmap_mode='r+')


X, y, splits = combine_split_data([x_train, x_valid], [y_train,y_valid])
check_data(X, y, splits)
batch_tfms = TSStandardize(by_sample=True, by_var=True)
'''

X, y, splits = get_UCR_data('NATOPS', split_data=False)
learn = TSClassifier(X, y, splits=splits, bs=[64, 128], batch_tfms=TSStandardize(by_sample=True), arch=InceptionTimePlus, metrics=accuracy, cbs=ShowGraph())
lr_max = learn.lr_find()
print(f'lf_max = {lr_max.valley}')

learn = TSClassifier(X, y, splits=splits, bs=[64, 128], batch_tfms=TSStandardize(by_sample=True), arch=InceptionTimePlus, metrics=accuracy, cbs=ShowGraph())
learn.fit_one_cycle(5, lr_max=lr_max.valley)

from optuna.integration import FastAIPruningCallback



def objective(trial:optuna.Trial):
    
    # Define search space here. More info here https://optuna.readthedocs.io/en/stable/tutorial/10_key_features/002_configurations.html
    nf = trial.suggest_categorical('num_filters', [32, 64, 96]) # search through all categorical values in the provided list
    depth = trial.suggest_int('depth', 3, 9, step=3) # search through all integer values between 3 and 9 with 3 increment steps
    dropout_rate = trial.suggest_float("dropout_rate", 0.0, 0.5, step=.1) # search through all float values between 0.0 and 0.5 with 0.1 increment steps
    learning_rate = trial.suggest_float("learning_rate", 1e-5, 1e-2, log=True)  # search through all float values between 0.0 and 0.5 in log increment steps
    
    batch_tfms = TSStandardize(by_sample=True)
    learn = TSClassifier(X, y, splits=splits, bs=[64, 128], batch_tfms=batch_tfms,
                         arch=InceptionTimePlus, arch_config={'nf':nf, 'fc_dropout':dropout_rate, 'depth':depth},
                         metrics=accuracy, cbs=FastAIPruningCallback(trial))


    with ContextManagers([learn.no_logging(), learn.no_bar()]): # [Optional] this prevents fastai from printing anything during training
        learn.fit_one_cycle(5, lr_max=learning_rate)

    # Return the objective value
    return learn.recorder.values[-1][1] # return the validation loss value of the last epoch

study = optuna.create_study(direction='minimize')
study.optimize(objective, n_trials=100)
```

Let me know how you go. Many thanks in advance.

**Comment by nzw0301** [comment]
Thanks. I suppose it comes from `FastAIPruningCallback`'s pruning part. So you can ignore it.
